test/testutil/init.c

```diff
@@ -7,9 +7,135 @@
  * https://www.openssl.org/source/license.html
  */
 
+#include <string.h>
+#include <openssl/opensslconf.h>
+#include <openssl/trace.h>
+#include "apps.h"
 #include "../testutil.h"
 
+#ifndef OPENSSL_NO_TRACE
+typedef struct tracedata_st {
+    BIO *bio;
+    unsigned int ingroup:1;
+} tracedata;
+
+static size_t internal_trace_cb(const char *buf, size_t cnt,
+                                int category, int cmd, void *vdata)
+{
+    int ret = 0;
+    tracedata *trace_data = vdata;
+    union {
```

Comment:
Ugh.  What about a cast to unsigned long for the one place this is used?
Or are some thread id's not integral?

Reply:
Thread IDs aren't always integral.  You should read the [notes in the Linux man-page for `pthread_self`](http://man7.org/linux/man-pages/man3/pthread_self.3.html#NOTES)

Reply:
What about a hexdump of the entire thread_id?
A cast to unsigned long is a kludge :)

Reply:
What can I say, it's copied from over here:

https://github.com/openssl/openssl/blob/b720949ec09f4dfbff1becc75ca808997e5b51bd/apps/openssl.c#L128-L170

[sigh] no time for deeper refactoring today, but should probably be part of this PR after all.  I'll put this in WIP for the time being.